Add experts map to the repo

spastorino opened the pull request:
@nikomatsakis @davidtwco we agreed on adding a document here that links the experts map.

Now I wonder, shouldn't we do the opposite?, I mean, build a table in this repo and link from the hackmd to this place just in case people have the old link.

The downside is that it's a bit more involved to add/remove/change stuff on github because requires a PR. But I guess this is "stable" now?.

davidtwco commented on experts/MAP.md:
```diff
@@ -0,0 +1,4 @@
+# Compiler team expert map
```
```suggestion
# Expert Map
```

I don't think we need to say "compiler team", since that is implied by this repository.

davidtwco commented on experts/MAP.md:
```diff
@@ -0,0 +1,4 @@
+# Compiler team expert map
+
+This document containts different areas of the compiler and people that
```
```suggestion
This document contains the different areas of the compiler and the people that
```

davidtwco commented on experts/MAP.md:
```diff
@@ -0,0 +1,4 @@
+# Compiler team expert map
+
+This document containts different areas of the compiler and people that
+know a bit about each area. For now that document is on hackmd https://hackmd.io/Izvor8KZRiqUgcUyd2CYTw
```
```suggestion
are experts in each area. While the expert map is being fleshed out, [you can find it on HackMD][map].

[map]: https://hackmd.io/Izvor8KZRiqUgcUyd2CYTw
```

davidtwco commented:
If you could also link to this from the root `README.md` then that would be great!

spastorino commented:
@davidtwco I don't mind at all, you're helping me a lot with these kind of comments to improve my english a bit more ;). Have addressed all your comments and squashed all the commits to avoid a lot of the generated noise.

nikomatsakis commented on experts/MAP.md:
```diff
@@ -0,0 +1,6 @@
+# Expert Map
+
+This document contains the different areas of the compiler and the people that
+are experts in each area. While the expert map is being fleshed out, [you can find it on HackMD][map].
```
I would add:

---

This map is intended to help you figure out who you should ask for help if you have questions about how some area works.

spastorino commented:
@nikomatsakis addressed your comments

alexreg commented on README.md:
```diff
@@ -90,6 +90,11 @@ Name                                               | Leads
+## Expert Map
+
+The [experts directory](experts) contains a map of areas of the compiler
+to experts from the Rust compiler team.
```
The "experts directory" link seems to be broken (old pull request?).

davidtwco replied:
I think it is just copied from the PR diff where it would correctly link to the `experts` directory in this repository when rendered in a markdown document.
